src/main.rs

```diff
@@ -97,12 +97,36 @@ fn run_script(vm: &VirtualMachine, script_file: &str) -> PyResult {
     debug!("Running file {}", script_file);
     // Parse an ast from it:
     let file_path = Path::new(script_file);
-    match util::read_file(file_path) {
-        Ok(source) => _run_string(vm, &source, file_path.to_str().unwrap().to_string()),
-        Err(err) => {
-            error!("Failed reading file: {:?}", err.kind());
-            std::process::exit(1);
+    if file_path.is_file() {
+        match util::read_file(file_path) {
+            Ok(source) => _run_string(vm, &source, file_path.to_str().unwrap().to_string()),
```

Comment:
Here you duplicated this line of code. Change the code such that first the filename is determined, and then this line is executed. Like this:
```
let file_path = if file_path.is_file() {
  file_path
} else if file_path.is_dir() {
  let file_path = file_path.join("__main__.py");
  if file_path.is_file() {
     file_path
  } else {
    error("No __main__.py found...")
  }
} else {
};

match util::read_file(&file_path) {
  Ok => run_string() 
   ...
};